code/datums/traits/_quirk.dm

```diff
@@ -29,9 +29,10 @@
 	add()
 	if(spawn_effects)
 		on_spawn()
-		addtimer(CALLBACK(src, .proc/post_add), 30)
+	RegisterSignal(quirk_holder, COMSIG_MOB_LOGIN, .proc/post_add)
 
 /datum/quirk/Destroy()
+	UnregisterSignal(quirk_holder, COMSIG_MOB_LOGIN)
```

Comment:
```suggestion
```
Signals clear themselves properly already on deletion of either listener or emitter.